src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -2225,6 +2253,17 @@ private void ValidateOrganizationUpgradeParameters(Models.StaticStore.Plan plan,
     private async Task ValidateOrganizationUserUpdatePermissions(Guid organizationId, OrganizationUserType newType,
```

Comment:
Looking at this method again, it's actually related to the permissions of the user _doing the update_, not the _user being updated_. So this is not the right place to include this code snippet. (note that it's called from `UpdateUserGroupsAsync`, which doesn't have anything to do with changing a user's permissions. We don't actually need the check there) 

Please review and find somewhere else to put it. I don't really like adding more helper methods to this class, but that's probably what we're looking at in this PR.

Reply:
I ended up adding a new helper method `ValidateOrganizationCustomPermissionsEnabledAsync`. I've also added a few more unit tests around it to make sure it didn't break anything else.